src/test/kotlin/com/atlassian/performance/tools/virtualusers/LoadTestIT.kt

```diff
@@ -0,0 +1,160 @@
+import com.atlassian.performance.tools.infrastructure.api.jira.JiraLaunchTimeouts
+import com.atlassian.performance.tools.infrastructure.api.jira.JiraNodeConfig
+import com.atlassian.performance.tools.infrastructure.api.jvm.OracleJDK
+import com.atlassian.performance.tools.infrastructure.api.profiler.AsyncProfiler
+import com.atlassian.performance.tools.ssh.api.Ssh
+import com.atlassian.performance.tools.virtualusers.api.TemporalRate
+import com.atlassian.performance.tools.virtualusers.api.VirtualUserLoad
+import com.atlassian.performance.tools.virtualusers.api.VirtualUserOptions
+import com.atlassian.performance.tools.virtualusers.api.config.VirtualUserBehavior
+import com.atlassian.performance.tools.virtualusers.api.config.VirtualUserTarget
+import com.atlassian.performance.tools.virtualusers.lib.docker.execAsResource
+import com.atlassian.performance.tools.virtualusers.lib.infrastructure.FixedMysqlDatabase
+import com.atlassian.performance.tools.virtualusers.lib.infrastructure.FixedOracleJDK
+import com.atlassian.performance.tools.virtualusers.lib.infrastructure.FixedPublicJiraSoftwareDistribution
+import com.atlassian.performance.tools.virtualusers.lib.infrastructure.SshJiraNode
+import com.atlassian.performance.tools.virtualusers.lib.sshubuntu.SudoSshUbuntuContainer
+import com.atlassian.performance.tools.virtualusers.lib.sshubuntu.SudoSshUbuntuImage
+import com.github.dockerjava.api.model.ExposedPort
+import com.github.dockerjava.core.DockerClientBuilder
+import org.junit.Test
+import java.net.URI
+import java.time.Duration
+import java.util.*
+import java.util.concurrent.Executors
+
+class LoadTestIT {
+
+    private val dataset: Dataset = URI("https://s3-eu-central-1.amazonaws.com/")
+        .resolve("jpt-custom-datasets-storage-a008820-datasetbucket-dah44h6l1l8p/")
+        .resolve("jsw-7.13.0-100k-users-sync/")
+        .let { bucket ->
+            Dataset(
+                database = FixedMysqlDatabase(MySqlDatabase(
+                    HttpDatasetPackage(
+                        uri = bucket.resolve("database.tar.bz2"),
+                        downloadTimeout = Duration.ofMinutes(6)
+                    )
+                )),
+                jiraHomeSource = JiraHomePackage(HttpDatasetPackage(
+                    uri = bucket.resolve("jirahome.tar.bz2"),
+                    downloadTimeout = Duration.ofMinutes(6)
+                )),
+                label = "100k users"
+            )
+        }
+
+    private val behavior = VirtualUserBehavior.Builder(TracingScenario::class.java)
+        .createUsers(true)
+        .browser(LoadTestTest.TestBrowser::class.java)
+        .skipSetup(true)
+
+    @Test
+    fun shouldCreateUsersInParallelDespiteBigUserBase() {
+        val pool = Executors.newCachedThreadPool()
+        val nodes = 6
+        val load = VirtualUserLoad.Builder()
+            .virtualUsers(75)
+            .ramp(Duration.ofSeconds(15))
+            .flat(Duration.ofSeconds(50))
```

Comment:
WRT https://github.com/atlassian/virtual-users/pull/2#issuecomment-473918610
I'll try to hide the problem by prolonging the `VirtualUserLoad`, making our CI slower, but less flaky.
When we find a user generator that works reliably even for short-load tests, then we can revert the load to be shorter.
If prolonging doesn't work, it might still be worth merging, because it shows how we go from 100% failure rate to 10%.

Reply:
So I prolonged it and ran locally. It [worked 10/10 times](https://gist.github.com/dagguh/18c331adb2f40ab2f2ff7870d7e487f7).
Min time: 11m 19s
Max green time: 17m 33s

Interestingly, [CI failed](https://circleci.com/gh/dagguh/virtual-users/53). They have weaker hardware than my local machine, so the Dockerized Jira might make the user generation even slower. The symptom is similar as to the one we get when we only bump the HTTP read timeout. It's:
> Expected 12 VUs to still be active, but encountered 11

The [HTTP timeout one](https://circleci.com/gh/dagguh/virtual-users/52) is:
> Expected 12 VUs to still be active, but encountered 0

The class of the problem is the same, but the scale is different.

There's an alternative solution: do not create the users at all, but use a "list of known users". It's the good old "external knowledge" design, which is dataset-specific. We can easily open up SPI that would allow the users to choose between the solutions. For some cases the user generation will work fine (e.g. if there are not a lot of users or Jira is set up to use async user caches), for others it will be unacceptable (e.g. our CI or hardware exploration tests for weak hardware).

Reply:
Given the fact how big this PR is, I'll merge it as is. We can add the new SPI in a new PR.